Automattic/wp-calypso
# Jetpack: Remove obsolete isJetpackMinimumVersion checks

Currently we use `isJetpackMinimumVersion` in many locations, to display a feature or functionality only for a certain Jetpack version or newer. However, Jetpack's official support policy is to support the current - 1 version. Any older versions aren't supported. This means we can remove any checks that check for 7.9-only or older versions of Jetpack. This basically means removing all of the `isJetpackMinimumVersion` usages.

We keep the `isJetpackMinimumVersion` selector though, as it might be useful for future features.

This cleanup helps in removing some dead unnecessary code and keeping the codebase cleaner.

#### Changes proposed in this Pull Request

* Jetpack: Remove obsolete `isJetpackMinimumVersion` checks.

#### Testing instructions
Test the following screens and flows and verify they work like they did before:
* Simple payments dialog in classic editor (Jetpack and WP.com site).
* Memberships in the Simple Payments dialog in classic editor.
* Comments - navigation - newest / oldest sorting functionality.
* Comments - comment tree and comment nesting.
* Comments - comment inline editing.
* Comments - main screen.
* Tools -> Earn -> Payments - recurring payments.
* Tools -> Marketing -> Sharing Buttons -> Options card settings.
* Tools -> Marketing -> Connections -> Mailchimp connection.
* Tools -> Marketing - The "Get social, and share your blog posts where the people are" card.
* Plans - Jetpack backup purchase card for Jetpack sites.
* Plugins -> Upload (http://calypso.localhost:3000/plugins/upload/:site), test with single site and multisite (multisite should suggest going to wp-admin).
* Pages -> Edit Page (classic editor) -> Sidebar -> Page Attributes, selecting a parent page.
* Posts -> Edit Post (classic editor) -> Sidebar -> Categories & Tags, displaying categories with the current category selected and tags inputted.
* Posts -> Edit Post (classic editor) -> Sidebar - test everything on a Jetpack and WP.com site.
* Posts -> Edit Post (classic editor) -> Sidebar -> Categories & Tags, selecting a category and inputting tags.
* Not removing the selector because it will be necessary as we release new features.
